fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabase.java

```diff
@@ -238,6 +242,23 @@ public synchronized FDBLocalityProvider getLocalityProvider() {
         return localityProvider;
     }
 
+    /**
+     * Get the FDB API version that this database was created with. This can be used by calling
+     * code to determine what methods are safe to call, and for compatibility code to know how to
+     * properly execute operations that may require differently depending on the API. This is an
+     * internal method that should only be called by internal modules, as higher level Record
+     * Layer APIs should shield adopters from needing to know this configuration value except
+     * possibly during application initialization.
+     *
+     * @return the API version of this database
+     * @see APIVersion
+     * @see FDBDatabaseFactory#setAPIVersion(APIVersion)
+     */
+    @API(API.Status.INTERNAL)
+    public APIVersion getAPIVersion() {
```

Comment:
How about some convenience method, like `FDBTransactionContext.atLeastVersion(APIVersion)`? Or does that go against the idea of keeping this internal?

Reply:
I think that's a reasonable convenience method to have. I've marked the new methods as `INTERNAL` as well, under the spirit of keeping this internal, but it seems like a method we might want ourselves.